### PR TITLE
feat(otlp-sink): Add retry, batching, OkHttp client, and encoder refactor

### PR DESCRIPTION
**Pull request by huypham612**
### Description
This update introduces several enhancements to the OTLP sink:

    Retry Mechanism: Implements exponential backoff with jitter to gracefully handle transient failures (e.g. 429, 5xx).

    Batching Support: Adds configurable batching for efficient trace delivery.

    OkHttp Integration: Replaces basic HTTP logic with OkHttp client, enabling future async and connection reuse support.

    Encoder Refactor: Replaces OutputCodec usage with OTelProtoStandardCodec.OTelProtoEncoder for direct protobuf encoding of ExportTraceServiceRequest.

    Test Coverage: Improves unit test coverage to ~95% with additional edge case validation.

    E2E Testing: Performed end-to-end testing with real OTLP trace ingestion into AWS X-Ray.

    Sink Renaming: Renamed plugin to otlp sink for clarity and consistency.
### Issues Resolved
N/A
 
### Check List
- [X] New functionality includes testing.
- [X] New functionality has a documentation issue. Please link to it in this PR.
  - [X] New functionality has javadoc added
- [X] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-plugins/otel-proto-common/src/main/java/org/opensearch/dataprepper/plugins/otel/codec/OtlpTraceOutputCodec.java [hunk 1]
```diff
@@ -24,7 +24,7 @@
  * into OpenTelemetry Protocol (OTLP) binary format using protobuf.
  * <p>
  * This codec is primarily intended for use with trace data that will be
- * forwarded to systems such as AWS X-Ray via OTLP.
+ * forwarded to OTLP endpoint.
  */
 @DataPrepperPlugin(name = "otlp_trace", pluginType = OutputCodec.class)
 public class OtlpTraceOutputCodec implements OutputCodec {
```
Comment (geekgirl10) [inline data-prepper-plugins/otel-proto-common/src/main/java/org/opensearch/dataprepper/plugins/otel/codec/OtlpTraceOutputCodec.java:30]: Should we remove this Codec since OTLP Is Not Compatible with OtlpTraceOutputCodec.writeEvent()?

Comment (huypham612) [reply data-prepper-plugins/otel-proto-common/src/main/java/org/opensearch/dataprepper/plugins/otel/codec/OtlpTraceOutputCodec.java:30]: Yeah, let's hang on this for a bit, to get feedback from OSI. I don't want to remove your work prematurely.

We can remove it before the PR to data-prepper main.

**Review by geekgirl10 — APPROVED** [review]
